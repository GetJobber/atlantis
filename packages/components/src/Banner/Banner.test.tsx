import React from "react";
import renderer from "react-test-renderer";
import { cleanup, fireEvent, render } from "@testing-library/react";
import { Banner } from ".";

afterEach(cleanup);

it("renders a success banner", () => {
  const tree = renderer
    .create(<Banner type="success">Success</Banner>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders an error banner", () => {
  const tree = renderer.create(<Banner type="error">Fail</Banner>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders a notice banner", () => {
  const tree = renderer
    .create(<Banner type="notice">Notice me</Banner>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders a warning banner", () => {
  const tree = renderer.create(<Banner type="warning">Warn</Banner>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders without close button", () => {
  const { queryByLabelText } = render(
    <Banner type="warning" dismissible={false}>
      Foo
    </Banner>,
  );
  expect(queryByLabelText("Close")).toBeNull();
});

it("renders with close button", () => {
  const { queryByLabelText } = render(
    <Banner type="warning" dismissible={true}>
      Foo
    </Banner>,
  );
  expect(queryByLabelText("Close")).toBeTruthy();
});

test("it should call the handler with a number value", () => {
  const changeHandler = jest.fn();

  const { getByLabelText } = render(
    <Banner type="success" onDismiss={changeHandler}>
      Close me
    </Banner>,
  );

  fireEvent.click(getByLabelText("Close"));
  expect(changeHandler).toHaveBeenCalledTimes(1);
});

it("renders a banner with a primary action", () => {
  const tree = renderer
    .create(
      <Banner
        type="success"
        primaryAction={{
          label: "smash me",
        }}
      >
        Bruce
      </Banner>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders a banner with a primary 'learning' action when the type is 'notice'", () => {
  const tree = renderer
    .create(
      <Banner
        type="notice"
        primaryAction={{
          label: "smash me",
        }}
      >
        Bruce
      </Banner>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("it should call the onClick when primaryAction is present", () => {
  const onClick = jest.fn();

  const { getByText } = render(
    <Banner
      type="success"
      primaryAction={{
        label: "Hello World",
        onClick: () => onClick(),
      }}
    >
      Bruce
    </Banner>,
  );

  fireEvent.click(getByText("Hello World"));
  expect(onClick).toHaveBeenCalledTimes(1);
});
