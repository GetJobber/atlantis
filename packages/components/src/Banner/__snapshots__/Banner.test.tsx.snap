// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a banner with a primary 'learning' action when the type is 'notice' 1`] = `
<div
  className="flash notice"
  role="status"
>
  <div
    className="bannerContent dismissibleSpacing"
  >
    <p
      className="base regular base greyBlueDark"
    >
      Bruce
    </p>
    <div
      className="bannerAction"
    >
      <button
        className="button small learning tertiary"
        disabled={false}
        type="button"
      >
        <span
          className="base extraBold smaller uppercase lightBlue"
        >
          smash me
        </span>
      </button>
    </div>
  </div>
  <button
    aria-label="Close"
    className="closeButton"
    onClick={[Function]}
  >
    <svg
      className="ZDlJU6tECg7ouG-b27gya _3ctOeOMP7zLhU37n25xoZC"
      data-testid="cross"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        className="_1Fz3GVTXQVAtpcbXjBZ1DQ"
        d="M512 451.669l-225.835-225.835c-8.047-7.772-18.825-12.073-30.012-11.976s-21.888 4.585-29.799 12.495c-7.911 7.911-12.398 18.612-12.495 29.799s4.204 21.964 11.976 30.012l225.835 225.835-225.835 225.835c-7.772 8.047-12.073 18.825-11.976 30.012s4.585 21.888 12.495 29.798c7.91 7.91 18.612 12.399 29.799 12.497s21.965-4.203 30.012-11.977l225.835-225.835 225.835 225.835c8.047 7.774 18.825 12.075 30.012 11.977s21.888-4.587 29.798-12.497c7.91-7.91 12.399-18.611 12.497-29.798 0.094-11.187-4.203-21.965-11.977-30.012l-225.835-225.835 225.835-225.835c4.075-3.936 7.326-8.644 9.562-13.85s3.413-10.804 3.46-16.469c0.051-5.665-1.028-11.284-3.174-16.527s-5.312-10.007-9.318-14.013c-4.006-4.006-8.772-7.174-14.016-9.319-5.239-2.146-10.859-3.225-16.525-3.176s-11.264 1.226-16.469 3.462c-5.205 2.236-9.916 5.487-13.85 9.562l-225.835 225.835z"
      />
    </svg>
  </button>
</div>
`;

exports[`renders a banner with a primary action 1`] = `
<div
  className="flash success"
  role="status"
>
  <div
    className="bannerContent dismissibleSpacing"
  >
    <p
      className="base regular base greyBlueDark"
    >
      Bruce
    </p>
    <div
      className="bannerAction"
    >
      <button
        className="button small work tertiary"
        disabled={false}
        type="button"
      >
        <span
          className="base extraBold smaller uppercase green"
        >
          smash me
        </span>
      </button>
    </div>
  </div>
  <button
    aria-label="Close"
    className="closeButton"
    onClick={[Function]}
  >
    <svg
      className="ZDlJU6tECg7ouG-b27gya _3ctOeOMP7zLhU37n25xoZC"
      data-testid="cross"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        className="_1R4_alABGZICCj4KPfBOIa"
        d="M512 451.669l-225.835-225.835c-8.047-7.772-18.825-12.073-30.012-11.976s-21.888 4.585-29.799 12.495c-7.911 7.911-12.398 18.612-12.495 29.799s4.204 21.964 11.976 30.012l225.835 225.835-225.835 225.835c-7.772 8.047-12.073 18.825-11.976 30.012s4.585 21.888 12.495 29.798c7.91 7.91 18.612 12.399 29.799 12.497s21.965-4.203 30.012-11.977l225.835-225.835 225.835 225.835c8.047 7.774 18.825 12.075 30.012 11.977s21.888-4.587 29.798-12.497c7.91-7.91 12.399-18.611 12.497-29.798 0.094-11.187-4.203-21.965-11.977-30.012l-225.835-225.835 225.835-225.835c4.075-3.936 7.326-8.644 9.562-13.85s3.413-10.804 3.46-16.469c0.051-5.665-1.028-11.284-3.174-16.527s-5.312-10.007-9.318-14.013c-4.006-4.006-8.772-7.174-14.016-9.319-5.239-2.146-10.859-3.225-16.525-3.176s-11.264 1.226-16.469 3.462c-5.205 2.236-9.916 5.487-13.85 9.562l-225.835 225.835z"
      />
    </svg>
  </button>
</div>
`;

exports[`renders a notice banner 1`] = `
<div
  className="flash notice"
  role="status"
>
  <div
    className="bannerContent dismissibleSpacing"
  >
    <p
      className="base regular base greyBlueDark"
    >
      Notice me
    </p>
  </div>
  <button
    aria-label="Close"
    className="closeButton"
    onClick={[Function]}
  >
    <svg
      className="ZDlJU6tECg7ouG-b27gya _3ctOeOMP7zLhU37n25xoZC"
      data-testid="cross"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        className="_1Fz3GVTXQVAtpcbXjBZ1DQ"
        d="M512 451.669l-225.835-225.835c-8.047-7.772-18.825-12.073-30.012-11.976s-21.888 4.585-29.799 12.495c-7.911 7.911-12.398 18.612-12.495 29.799s4.204 21.964 11.976 30.012l225.835 225.835-225.835 225.835c-7.772 8.047-12.073 18.825-11.976 30.012s4.585 21.888 12.495 29.798c7.91 7.91 18.612 12.399 29.799 12.497s21.965-4.203 30.012-11.977l225.835-225.835 225.835 225.835c8.047 7.774 18.825 12.075 30.012 11.977s21.888-4.587 29.798-12.497c7.91-7.91 12.399-18.611 12.497-29.798 0.094-11.187-4.203-21.965-11.977-30.012l-225.835-225.835 225.835-225.835c4.075-3.936 7.326-8.644 9.562-13.85s3.413-10.804 3.46-16.469c0.051-5.665-1.028-11.284-3.174-16.527s-5.312-10.007-9.318-14.013c-4.006-4.006-8.772-7.174-14.016-9.319-5.239-2.146-10.859-3.225-16.525-3.176s-11.264 1.226-16.469 3.462c-5.205 2.236-9.916 5.487-13.85 9.562l-225.835 225.835z"
      />
    </svg>
  </button>
</div>
`;

exports[`renders a success banner 1`] = `
<div
  className="flash success"
  role="status"
>
  <div
    className="bannerContent dismissibleSpacing"
  >
    <p
      className="base regular base greyBlueDark"
    >
      Success
    </p>
  </div>
  <button
    aria-label="Close"
    className="closeButton"
    onClick={[Function]}
  >
    <svg
      className="ZDlJU6tECg7ouG-b27gya _3ctOeOMP7zLhU37n25xoZC"
      data-testid="cross"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        className="_1R4_alABGZICCj4KPfBOIa"
        d="M512 451.669l-225.835-225.835c-8.047-7.772-18.825-12.073-30.012-11.976s-21.888 4.585-29.799 12.495c-7.911 7.911-12.398 18.612-12.495 29.799s4.204 21.964 11.976 30.012l225.835 225.835-225.835 225.835c-7.772 8.047-12.073 18.825-11.976 30.012s4.585 21.888 12.495 29.798c7.91 7.91 18.612 12.399 29.799 12.497s21.965-4.203 30.012-11.977l225.835-225.835 225.835 225.835c8.047 7.774 18.825 12.075 30.012 11.977s21.888-4.587 29.798-12.497c7.91-7.91 12.399-18.611 12.497-29.798 0.094-11.187-4.203-21.965-11.977-30.012l-225.835-225.835 225.835-225.835c4.075-3.936 7.326-8.644 9.562-13.85s3.413-10.804 3.46-16.469c0.051-5.665-1.028-11.284-3.174-16.527s-5.312-10.007-9.318-14.013c-4.006-4.006-8.772-7.174-14.016-9.319-5.239-2.146-10.859-3.225-16.525-3.176s-11.264 1.226-16.469 3.462c-5.205 2.236-9.916 5.487-13.85 9.562l-225.835 225.835z"
      />
    </svg>
  </button>
</div>
`;

exports[`renders a warning banner 1`] = `
<div
  className="flash warning"
  role="status"
>
  <div
    className="bannerContent dismissibleSpacing"
  >
    <p
      className="base regular base greyBlueDark"
    >
      Warn
    </p>
  </div>
  <button
    aria-label="Close"
    className="closeButton"
    onClick={[Function]}
  >
    <svg
      className="ZDlJU6tECg7ouG-b27gya _3ctOeOMP7zLhU37n25xoZC"
      data-testid="cross"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        className="_2QH3YFqLKdR7aDlgrj2Hh6"
        d="M512 451.669l-225.835-225.835c-8.047-7.772-18.825-12.073-30.012-11.976s-21.888 4.585-29.799 12.495c-7.911 7.911-12.398 18.612-12.495 29.799s4.204 21.964 11.976 30.012l225.835 225.835-225.835 225.835c-7.772 8.047-12.073 18.825-11.976 30.012s4.585 21.888 12.495 29.798c7.91 7.91 18.612 12.399 29.799 12.497s21.965-4.203 30.012-11.977l225.835-225.835 225.835 225.835c8.047 7.774 18.825 12.075 30.012 11.977s21.888-4.587 29.798-12.497c7.91-7.91 12.399-18.611 12.497-29.798 0.094-11.187-4.203-21.965-11.977-30.012l-225.835-225.835 225.835-225.835c4.075-3.936 7.326-8.644 9.562-13.85s3.413-10.804 3.46-16.469c0.051-5.665-1.028-11.284-3.174-16.527s-5.312-10.007-9.318-14.013c-4.006-4.006-8.772-7.174-14.016-9.319-5.239-2.146-10.859-3.225-16.525-3.176s-11.264 1.226-16.469 3.462c-5.205 2.236-9.916 5.487-13.85 9.562l-225.835 225.835z"
      />
    </svg>
  </button>
</div>
`;

exports[`renders an error banner 1`] = `
<div
  className="flash error"
  role="status"
>
  <div
    className="bannerContent dismissibleSpacing"
  >
    <p
      className="base regular base greyBlueDark"
    >
      Fail
    </p>
  </div>
  <button
    aria-label="Close"
    className="closeButton"
    onClick={[Function]}
  >
    <svg
      className="ZDlJU6tECg7ouG-b27gya _3ctOeOMP7zLhU37n25xoZC"
      data-testid="cross"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        className="_3kicqC60DO19i9GJUdTy7F"
        d="M512 451.669l-225.835-225.835c-8.047-7.772-18.825-12.073-30.012-11.976s-21.888 4.585-29.799 12.495c-7.911 7.911-12.398 18.612-12.495 29.799s4.204 21.964 11.976 30.012l225.835 225.835-225.835 225.835c-7.772 8.047-12.073 18.825-11.976 30.012s4.585 21.888 12.495 29.798c7.91 7.91 18.612 12.399 29.799 12.497s21.965-4.203 30.012-11.977l225.835-225.835 225.835 225.835c8.047 7.774 18.825 12.075 30.012 11.977s21.888-4.587 29.798-12.497c7.91-7.91 12.399-18.611 12.497-29.798 0.094-11.187-4.203-21.965-11.977-30.012l-225.835-225.835 225.835-225.835c4.075-3.936 7.326-8.644 9.562-13.85s3.413-10.804 3.46-16.469c0.051-5.665-1.028-11.284-3.174-16.527s-5.312-10.007-9.318-14.013c-4.006-4.006-8.772-7.174-14.016-9.319-5.239-2.146-10.859-3.225-16.525-3.176s-11.264 1.226-16.469 3.462c-5.205 2.236-9.916 5.487-13.85 9.562l-225.835 225.835z"
      />
    </svg>
  </button>
</div>
`;
